docs/dg/dev/guidelines/performance-guidelines/database-performance-quidelines.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Performance Guidance -  Batch Processing Traits in Propel ORM
+description: Learn how to optimize Propel ORM performance using ActiveRecordBatchProcessorTrait and CascadeActiveRecordBatchProcessorTrait for efficient batch processing, reduced database load, and support for complex entity relationships. Includes usage examples and best practices.
+last_updated: Jun 20, 2025
+template: concept-topic-template
+related:
+  - title: General performance guidelines
+    link: docs/dg/dev/guidelines/performance-guidelines/general-performance-guidelines.html
+---
+
+## Performance Guidance: Batch Processing Traits in Propel ORM
+
+### Prerequirements
+Update your Propel package to the: [spryker/propel:^3.45.0](https://github.com/spryker/propel/releases/tag/3.45.0)
+
+### ActiveRecordBatchProcessorTrait
+- This trait provides performance optimizations for database operations by allowing you to save Propel entities in batches.
+- It enables saving multiple records in a single operation, helping to resolve the - https://docs.spryker.com/docs/dg/dev/troubleshooting/troubleshooting-performance-issues/n+1-problem.html
+- It reduces database round trips and improves overall performance.
+
+#### Basic Usage Example:
+``` php
+class EntityManager {
+    use ActiveRecordBatchProcessorTrait;
+
+    public function processItems(array $itemTransfers): void {
+        // Load or create all entities at once
+        $items = $this->findOrCreateEntities($itemTransfers);
+
+        // Queue entities for batch processing
+        foreach ($items as $item) {
+            $this->persist($item);
+        }
+
+        // Save all collected entities in a single batch
+        $this->commit();
+    }
+}
+```
+#### Key Methods:
+- `persist($entity)`: Adds an entity to the batch queue. Handles separation for insert and update operations.
+- `commit()`: Commits all queued entities to the database in a single batch.
+
+### CascadeActiveRecordBatchProcessorTrait
+- Extends `ActiveRecordBatchProcessorTrait` by adding recursive cascade-saving functionality for related entities. 
+- Helps manage complex entity relationships during batch processing. 
+- Works with `BatchEntityPostSaveInterface`, which enables organizing `postSave` events for related entities in a batch-safe manner.
+
+#### Basic Usage Example:
```

Comment:
Can you please explain what is the use case (what are you trying to achieve) for the below code?